appcli/variables_manager.py

```diff
@@ -95,34 +98,119 @@ def set_all_variables(self, variables: Dict):
+                config_variables |= self.__convert_yaml_to_dict(
+                    data_string, config_file.stem
+                )
+            except Exception as ex:
+                raise Exception(
+                    f"Could not read configuration file at [{config_file}]"
+                ) from ex
+        return config_variables
 
     def __save(self, variables: Dict):
         """Saves the supplied Dict of variables to the configuration file
 
         Args:
-            variables (Dict): the variables to save
+            variables (Dict): All the variables to save (main config dictionary will be extracted).
         """
+        variables = variables[self.configuration_file.stem]
         full_path = self.configuration_file.absolute().as_posix()
         logger.debug(f"Saving configuration to [{full_path}] ...")
         with open(full_path, "w") as config_file:
             self.yaml.dump(variables, config_file)
+
+    def __read_configuration_source(self, configuration_filename: Path) -> str:
+        """Reads the given configuration file and returns its contents as a string
+
+        Args:
+            configuration_filename (Path): The location of the file to read from.
+
+        Returns:
+            str: A utf-8 encoded string of the files contents.
+
+        """
+        return self.configuration_file.read_text(encoding="utf-8")
+
+    def __convert_jinja_to_yaml(self, jinja_source: str, variables: Dict) -> str:
+        """Loads configuration data from a jinja2 string and applies
+            the provided templates.
+
+        Args:
+            jinja_source (str): A utf-8 encoding of the contents of the jinja2 file.
+            variables (Dict): Variables used to populate the template.
+
+        Returns:
+            str: A yaml string that has been templated with the provided variables.
+        """
+        template = Template(
+            jinja_source,
+            undefined=StrictUndefined,
+            trim_blocks=True,
+            lstrip_blocks=True,
+        )
+        return template.render(variables)
+
+    def __convert_yaml_to_dict(self, yaml_source: str, namespace: str) -> Dict:
+        """Loads configuration data from a provided yaml string.
+
+        Args:
+            yaml_source (str): A utf-8 encoding of some yaml data.
+            namespace (str): A unique namespace for this set of data.
+
+        Returns:
+            Dict: Configuration data from the file.
+        """
+        if not re.compile(r"[a-zA-Z0-9_]+").match(namespace):
+            raise Exception(f"[{namespace}] conatins invalid characters.")
+        yaml_data: dict = {namespace: self.yaml.load(yaml_source)}
+
+        # If the file is empty, the YAML library will load as `None`. Since
+        # we expect this function to return a valid dict, we return an
+        # empty dictionary if it's empty.
+        if yaml_data is None:
+            return {}
```

Comment:
If you are adding the namespace, then this will never evaluate to `True`

Reply:
Fixed.